src/Core/Metadata/MetadataController.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#nullable enable
+using System;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Quantum.IQSharp
+{
+
+    /// <summary>
+    ///     Raised when the available metadata has changed (e.g.: the client
+    ///     sends its user agent to the kernel).
+    /// </summary>
+    /// <param name="propertyName">
+    ///     Name of the metadata property whose value changed. May possibly
+    ///     be <c>null</c> if the property name is not known.
+    /// </param>
+    public delegate void OnMetadataChanged(string? propertyName);
```

Comment:
Add the sender of the event to make it easier on the subscriber to get the value that was changed.
I forgot to mentioned this as part of the WPF property change pattern...
OnMetadataChanged(IMetadataController sender, string? propertyName);